examples/2016-10-31/api_multiple_origin_cors/cors-util.js

```diff
@@ -0,0 +1,65 @@
+const DEFAULT_ALLOWED_HEADERS = [
+    "Content-Type",
+    "X-Amz-Date",
+    "Authorization",
+    "X-Api-Key",
+    "X-Amz-Security-Token"
+];
+
+/**
+ * Return an object literal that contains an Access-Control-Allow-Origin header
+ * if the request origin matches a pattern for an allowed origin.
+ * Otherwise, return an empty object literal.
+ * @param event Lambda event event associated with a request
+ * @param allowedOrigins A list of strings or regexes representing allowed origin URLs
+ * @return {Object} an object containing a header => value mapping, or nothing
+ */
+exports.makeOriginHeader = (event, allowedOrigins) => {
+    // determine origin
+    const origin = event.headers.Origin || event.headers.origin;
+    if (typeof origin === "undefined")
+        return {}; // no CORS headers necessary; browser will load resource
+
+    // look for origin in list of allowed origins
+    const allowedPattern = allowedOrigins.find(pattern => origin.match(pattern));
+    if (typeof allowedPattern === "undefined")
+        return {}; // return no CORS headers; browser will not load resource
+                   // we do not return a "null" origin because this is exploitable
+    
+    allowedOrigin = origin.match(allowedPattern)[0]; // determine exact match
+
+    // console.log("origin: " + origin);
+    // console.log("allowed by: " + allowedPattern);
+    
+    // produce origin header
+    return {"Access-Control-Allow-Origin": allowedOrigin};
```

Comment:
You mention `null` is exploitable. Maybe add a final `if (!allowedOrigin) return {}` to be safe.